### PR TITLE
Safer Contact Validation + Fix Null Reference

### PR DESCRIPTION
**Pull request by bookchin**
- Fix null ref callback in `shard#destroy` 
- Better contact validation, prevent crash on bad contact

### CODE REVIEW

File: lib/network/protocol.js [hunk 1]
```diff
@@ -337,7 +337,7 @@ Protocol.prototype.handleMirror = function(params, callback) {
       if (item.hash !== utils.rmd160(hasher256.digest())) {
         self._logger.warn('mirror read integrity check failed, destroying');
         report.end(ExchangeReport.FAILURE, 'FAILED_INTEGRITY');
-        item.shard.destroy();
+        item.shard.destroy(utils.noop);
```
Comment (braydonf) [inline lib/network/protocol.js]: Should we log warning if there was an error on the [callback](https://github.com/Storj/kfs/blob/master/lib/write-stream.js#L63)?

**Comment by braydonf** [comment]
I'm was tempted to try regex for isValidUrl, but running some benchmarks, it looks pretty quick regardless:
```
isValidUrl(true) x 96,376 ops/sec ±1.01% (181 runs sampled)
isValidUrl(false) x 92,015 ops/sec ±0.77% (182 runs sampled)
```

File: lib/network/index.js [hunk 2]
```diff
@@ -368,22 +367,13 @@ Network.prototype.subscribe = function(topic, handler) {
  * @param {Function} callback - Called on connection or error
  */
 Network.prototype.connect = function(uri, callback) {
-  return this.node.connect(this._createContact(uri), callback);
-};
+  callback = typeof callback === 'function' ? callback : utils.noop;
```
Comment (braydonf) [inline lib/network/index.js]: Also, should default callback log on error here?

**Comment by bookchin** [comment]
@braydonf okay, added logging warnings on callbacks

**Comment by braydonf** [comment]
LGTM

**Pull request merged**